src/conda_project/project.py

```diff
@@ -311,8 +311,8 @@ def environments(self) -> BaseEnvironments:
         env_paths = os.environ.get("CONDA_PROJECT_ENVS_PATH", "").split(os.pathsep)
 
         for path in env_paths:
-            if os.access(path, os.W_OK):
-                env_path = Path(path)
+            if os.access(path, os.W_OK) or not os.path.exists(path):
+                env_path = Path(os.path.join(self.directory, path))
```

Comment:
@AlbertDeFusco This should handle always making the prefix an absolute path (and you can take a look at the relative path test below, which verifies the prefix).

Reply:
Could we do it like this?

```python
env_path = self.directory / path
```

since `self.directory` is already a Path

Reply:
actually, i think there might be something missing with relative and absolute handling. For the case


```python
CONDA_PROJECT_ENVS_PATH=/opt/conda/envs:conda_envs:/home/.conda/envs
```

maybe this is a little extra explicit and there could be a shorter variant

```python
for path in env_paths:
    if not path:
        continue

    ## first setup the Path object
    if path.startswith('/'):
        path = Path(path)
    else:
        # this helps cover an edge case where your project directory is not writable
        # but there may be a writable location elsewhere in the env var
        path = self.directory / path

    if path.exists():
        if os.access(path, os.W_OK):
            env_path = path
            break
        else:
            continue
    else:
        if os.access(path.parent, os.W_OK):
            env_path = path
            break
        else:
            continue

envs = OrderedDict()
for env_name, sources in self._project_file.environments.items():
    envs[env_name] = Environment(
        name=env_name,
        sources=tuple([self.directory / str(s) for s in sources]),
        prefix=(env_path / env_name).absolute(),
        lockfile=self.directory / f"conda-lock.{env_name}.yml",
        project=weakref.proxy(self),
    )
Environments = create_model(
    "Environments",
    **{k: (Environment, ...) for k in envs},
    __base__=BaseEnvironments,
)
return Environments(**envs)
```

Reply:
Updated + added tests to handle the cases where a non-existent path's parent isn't writable, and where the project directory itself is not writable but a path in the environment variable is.